code/game/objects/items/weapons/storage/bluespace.dm

```diff
@@ -0,0 +1,63 @@
+/*
+ *		Bag of Holding
+ *		Miniature black hole
+*/
+
+/obj/item/weapon/storage/backpack/holding
+	name = "Bag of Holding"
+	desc = "A backpack that opens into a localized pocket of Blue Space."
+	origin_tech = "bluespace=4"
+	item_state = "holdingpack"
+	icon_state = "holdingpack"
+	fits_max_w_class = 4
+	max_combined_w_class = 28
+
+/obj/item/weapon/storage/backpack/holding/miniblackhole
+	name = "miniature black hole"
+	desc = "A miniature black hole that opens into a localized pocket of Blue Space."
+	icon_state = "porthole"
+	slot_flags = 0 //doesn't fit on your back!
+	w_class = 2 //fits in pockets!
+
+/obj/item/weapon/storage/backpack/holding/suicide_act(mob/user)
+	user.visible_message("<span class = 'danger'><b>[user] puts \the [src.name] on \his head and stretches the bag around \himself. With a sudden snapping sound, the bag shrinks to its original size, leaving no trace of [user].</b></span>")
+	user.drop_item(src)
+	qdel(user)
+
+/obj/item/weapon/storage/backpack/holding/miniblackhole/suicide_act(mob/user)
+	user.visible_message("<span class = 'danger'><b>[user] puts \the [src.name] on the ground and jumps inside, never to be seen again.<</b></span>")
+	user.drop_item(src)
+	qdel(user)
+
+/obj/item/weapon/storage/backpack/holding/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(W == src)
+		return // HOLY FUCKING SHIT WHY STORAGE CODE, WHY - pomf
+	var/obj/item/weapon/storage/backpack/holding/H = locate(/obj/item/weapon/storage/backpack/holding) in W
+	if(H || istype(W, /obj/item/weapon/storage/backpack/holding))
+		singulocreate(H, user)
+	. = ..()
+
+//BoH+BoH=Singularity, WAS commented out
+/obj/item/weapon/storage/backpack/holding/proc/singulocreate(var/obj/item/weapon/storage/backpack/holding/H, var/mob/user)
+	user.Weaken(10)
```

Comment:
Aaaaa why would you use this solution
Then you can still do it with hulk or by having a subverted borg dragging you or something
Now you also have the discrepancy of baguloth on the floor instagibbing you, but baguloth in your hands not doing the same